src/xenia/gpu/dxbc_shader_translator.cc

```diff
@@ -518,11 +525,18 @@ void DxbcShaderTranslator::StartVertexOrDomainShader() {
         a_.OpMov(uses_register_dynamic_addressing ? dxbc::Dest::X(0, 0, 0b0110)
                                                   : dxbc::Dest::R(0, 0b0110),
                  dxbc::Src::VDomain(0b010000));
-        // Copy the primitive index to r0.x as a float.
+        // Copy the primitive index + offset to r0.x as a float.
         uint32_t primitive_id_temp =
             uses_register_dynamic_addressing ? PushSystemTemp() : 0;
         in_primitive_id_used_ = true;
-        a_.OpUToF(dxbc::Dest::R(primitive_id_temp, 0b0001), dxbc::Src::VPrim());
+        a_.OpMov(dxbc::Dest::R(primitive_id_temp, 0b0001), dxbc::Src::VPrim());
+        a_.OpIAdd(dxbc::Dest::R(primitive_id_temp, 0b0001),
+                  dxbc::Src::R(primitive_id_temp, dxbc::Src::kXXXX),
+                  dxbc::Src::CB(cbuffer_index_system_constants_,
```

Comment:
System constants should be loaded via `LoadSystemConstant` (if I recall correctly), so it is ensured that the system constant buffer binding exists, and that the needed constant is marked as used in the resource definitions.